python/tritonserver/_api/_model.py

```diff
@@ -180,40 +181,52 @@ def async_infer(
         if inference_request is None:
             inference_request = InferenceRequest(model=self, **kwargs)
 
-        if (inference_request.response_queue is not None) and (
-            not isinstance(inference_request.response_queue, asyncio.Queue)
-        ):
-            raise InvalidArgumentError(
-                "asyncio.Queue must be used for async response iterator"
-            )
-
+        # The 'request' is referencing pointers managed by the 'inference_request', so
+        # the 'inference_request' needs to be kept alive as long as the 'request' is
+        # alive.
         request = inference_request._create_tritonserver_inference_request()
 
-        response_iterator = AsyncResponseIterator(
-            self,
-            request,
-            inference_request.response_queue,
-            raise_on_error,
-        )
+        self._server.infer_async(request)
```

Comment:
how do we pass in the requested memory type to the C++ side?

Reply:
I think we can add a new binding function to the request wrapper that allows setting a requested output memory type to the request wrapper object, then the response allocation function can access the requested memory type from the request wrapper object and allocate the correct memory type.

Reply:
let's make sure the stories to follow up on customizing allocation are in place and scheduled. Priority would be to specify the memory type (device id, etc.)

Reply:
yes, DLIS-7824.